pkg/operation/care/health.go

```diff
@@ -292,7 +290,14 @@ func (h *Health) checkControlPlane(
 		lokiEnabled = *h.gardenletConfiguration.Logging.Loki.Enabled
 	}
 
-	if gardenletfeatures.FeatureGate.Enabled(features.Logging) {
+	var loggingEnabled = false
+	if h.gardenletConfiguration != nil &&
+		h.gardenletConfiguration.Logging != nil &&
+		h.gardenletConfiguration.Logging.Enabled != nil {
+		loggingEnabled = *h.gardenletConfiguration.Logging.Enabled
+	}
+
+	if loggingEnabled {
```

Comment:
```suggestion
	if exitCondition, err := checker.CheckLoggingControlPlane(h.shoot.SeedNamespace, 
            h.shoot.IsLoggingEnabled() && loggingEnabled , lokiEnabled, condition, 
             seedStatefulSetLister); err != nil || exitCondition != nil {
			return exitCondition, err
	}
```
`h.shoot.IsLoggingEnabled()` and  `h.shoot.Purpose == gardencorev1beta1.ShootPurposeTesting` are the same.

Reply:
Thanks, I have already made the change in 2c706bc5bb102f9dee3a717b07faa38e725c14af.

Reply:
No, you haven't. Just remove `h.shoot.Purpose == gardencorev1beta1.ShootPurposeTesting` from the  `checker.CheckLoggingControlPlane()` call. Then replace it with `h.shoot.IsLoggingEnabled() && loggingEnabled` and then you could remove the first `if` enclosure.